.circleci/config.yml

```diff
@@ -141,6 +141,32 @@ jobs:
       - run:
           name: Codecov
           command: bash -i -c 'npx codecov'
+      - run:
+          name: Install and run sonarscanner
+          command: |
+            if [[ -n "${CIRCLE_BRANCH}" ]] && [[ -n "${CIRCLE_PULL_REQUEST}" || "${CIRCLE_BRANCH}" == "develop" ]]
+            then
+              export SONAR_SCANNER_VERSION=4.7.0.2747
+              export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
+              curl --create-dirs -sSLo $HOME/.sonar/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-linux.zip
+              unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
+              export PATH=$SONAR_SCANNER_HOME/bin:$PATH
+              export SONAR_SCANNER_OPTS="-server"
+
+              if [ -n "${CIRCLE_PULL_REQUEST}" ]
+              then
+                pr=$(echo https://api.github.com/repos/${CIRCLE_PULL_REQUEST:19} | sed "s/\/pull\//\/pulls\//")
```

Comment:
I don't think you need the `:19`, you've already verified that $CIRCLE_PULL_REQUEST has a non-zero length on line 156. Nit-picky, but I was confused by it at first, and I was going to ask why you chose 19 as the default value before I realized it will never be used.

Reply:
I added the `:19` since CIRCLE_PULL_REQUEST is in the format `https://github.com/dockstore/dockstore-ui2/pull/1798` and I just needed the `dockstore/dockstore-ui2/pull/1798` part for `https://api.github.com/repos/dockstore/dockstore-ui2/pulls/1798`.

Reply:
Whoops, I misread that as `${CIRCLE_PULL_REQUEST:=19}`, which evaluates to `19` if `CIRCLE_PULL_REQUEST` is not set.